tests/test_data/test_datasets.py

```diff
@@ -352,6 +374,14 @@ def test_tetrahedral_dataset(tmp_path, ds_name):
     assert np.all(tet_grid._vtk_offsets == np.array([0, 4, 8]))
     assert tet_grid.name == ds_name
 
+    # since vtk is attempted to be imported during a dependent function call
+    # let's call one of those so that vtk dict is properly populated
+    try:
```

Comment:
what is the reason behind using a `try / except` here? (when would vtk not be imported in the test?) maybe it could make sense to make this a `pytest.mark.parameterized()` where we run the function once with vtk and once without? maybe I'm just a bit confused about what this block of code is doing here.

Reply:
Revised this a bit. So, currently it works the following way:
- This test is designed to work for both when `vtk` is present and when it is not. 
- Which branch to check is determined by parameter `no_vtk`.
- I tried using `pytest.mark.parameterized()` to conveniently test both cases, but I couldn't make it work. it seems to be tricky to properly unload already loaded module. So, instead I just re-run the same tests in a separate pytest call [here](https://github.com/flexcompute/tidy3d/pull/1394/files#diff-002670f0bd6def8a37a0f88151b0e337d4d89b38dd18c46b32ea8bda1431c905R9) with a monkeypatch to skip loading `vtk` and `no_vtk=True`

Reply:
I see, I think the new implementation is fine.

btw I ran into something similar with wanting to test `adjoint` with and without jax installed, here's what I came up with in the end.

https://github.com/flexcompute/tidy3d/blob/pre/2.6/tests/test_plugins/test_adjoint.py#L1711-L1741

(I guess I also didn't use parameterize, maybe for similar reasons)